src/main/java/cd/go/authentication/ldap/Constants.java

```diff
@@ -18,17 +18,18 @@
 
 import com.thoughtworks.go.plugin.api.GoPluginIdentifier;
 
-import java.util.Collections;
+import java.util.Arrays;
 
 public interface Constants {
     // The type of this extension
     String EXTENSION_TYPE = "authorization";
 
     // The extension point API version that this plugin understands
     String API_VERSION = "1.0";
+    String NEW_API_VERSION = "2.0";
 
     // the identifier of this plugin
-    GoPluginIdentifier PLUGIN_IDENTIFIER = new GoPluginIdentifier(EXTENSION_TYPE, Collections.singletonList(API_VERSION));
+    GoPluginIdentifier PLUGIN_IDENTIFIER = new GoPluginIdentifier(EXTENSION_TYPE, Arrays.asList(API_VERSION, NEW_API_VERSION));
```

Comment:
Why do we need to support version "1.0"?

Reply:
@maheshp -- the contract didn't change between `v1` and `v2` extension. Even with `v2` changeset, extension `v1` will work.

Imagine a situation where we add a new bugfix on the master branch, if we say we only support `v2`, users using older plugin jar supporting `v1` extension will not get that fix.

---

I look at it as something getting for free. Going forward if we require to change plugin code to support both the versions, we can drop support for `v1`. :)

Reply:
> Imagine a situation where we add a new bugfix on the master branch, if we say we only support `v2`, users using older plugin jar supporting `v1` extension will not get that fix.

Sorry, I dint completely get it. Since the plugin is bundled I was wondering if this scenario exists, anyways we can talk later.

Reply:
@GaneshSPatil - This is bundled plugin and we are ensuring the compatibility with the latest version of GoCD. Even there is a chance of improving the plugin it will be fixed in coming GoCD release. So I don't think so we have to provide support for V1 and V2 both.

Reply:
@maheshp @bdpiparva -- yes yes. you're right. As we bundle this plugin, we don't need to support multiple versions. :)